File: simple/pythonvulnweb/cmd_exec/sub_class_test.py
from common.vuln_child import VulnChild

class CmdExecSubClassTest(VulnChild):
    def execute(self, cmd):
        # 命令执行漏洞测试：子类继承VulnChild，调用父类cmd_exec方法
        # 调用链：外部参数cmd -> CmdExecSubClassTest -> VulnChild -> VulnParent.cmd_exec -> os.system
        # 漏洞点：os.system执行用户控制的cmd参数
        return self.cmd_exec(cmd)
    
    def hardcoded_test(self):
        # 命令执行漏洞测试：硬编码场景，无外部参数
        # 调用链：hardcoded_test -> VulnChild -> VulnParent.hardcoded_cmd_exec -> os.system
        return self.hardcoded_cmd_exec()

File: simple/csvulnweb/CmdExec/SubClassTest.cs
using csvulnweb.Common;

namespace csvulnweb.CmdExec
{
    /// <summary>
    /// 命令执行漏洞 - 纯子类继承调用测试
    /// 调用链：请求参数 → VulnChild → VulnParent.VulnerableCmdExecWindows → 漏洞触发点
    /// </summary>
    public class CmdExecSubClassTest : VulnChild
    {
        /// <summary>
        /// 子类新增的命令执行方法，内部调用父类漏洞方法
        /// 调用链：外部参数 → TestCmdExec → CallCmdExec(VulnChild) → VulnerableCmdExecWindows(VulnParent)
        /// </summary>
        public string TestCmdExec(string userInput)
        {
            return CallCmdExec(userInput);
        }

        /// <summary>
        /// 子类硬编码漏洞测试方法
        /// 调用链：TestHardcodedCmdExec → CallHardcodedCmdExec(VulnChild) → HardcodedCmdExec(VulnParent)
        /// </summary>
        public string TestHardcodedCmdExec()
        {
            return CallHardcodedCmdExec();
        }
    }
}
